Trainer_v4/TrainerBehaviour.cs

```diff
@@ -362,7 +362,7 @@ private void Update()
 			GameSettings.MaxFloor = 100; //10 default
 			AI.MaxBoxes = Helpers.GetProperty(TrainerSettings, "IncreaseCourierCapacity") ? 108 : 54;
 			AI.BoxPrice = Helpers.GetProperty(TrainerSettings, "ReduceBoxPrice") ? 62.5f : 125;
-			Settings.Environment.ISPCostFactor = Helpers.GetProperty(TrainerSettings, "ReduceISPCost") ? 15f : 30f;
+			Settings.Environment.ISPCostFactor = 0f;
```

Comment:
This will force all players to have this option enabled if they have the trainer activated. You need to store the current environment ISPCostFactor value on game load and then if a player has the ReduceISPCost option enabled, set it to zero or some other value and if the player turns off that option you need to restore the original value to it.